src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponentContext.cs

```diff
@@ -2,25 +2,58 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.IO;
-using System.Reflection;
 using Microsoft.Framework.Internal;
 
 namespace Microsoft.AspNet.Mvc
 {
+    /// <summary>
+    /// A context for View Components.
+    /// </summary>
     public class ViewComponentContext
     {
-        public ViewComponentContext([NotNull] TypeInfo componentType, [NotNull] ViewContext viewContext,
+        /// <summary>
+        /// Creates a new <see cref="ViewComponentContext"/>.
+        /// </summary>
+        /// <param name="viewComponentDescriptor">
+        /// The <see cref="ViewComponentContext"/> for the View Component being invoked.
+        /// </param>
+        /// <param name="arguments">The View Component arguments.</param>
+        /// <param name="viewContext">The <see cref="ViewContext"/>.</param>
+        /// <param name="writer">The <see cref="TextWriter"/> for writing output.</param>
+        public ViewComponentContext(
+            [NotNull] ViewComponentDescriptor viewComponentDescriptor,
+            [NotNull] object[] arguments,
+            [NotNull] ViewContext viewContext,
             [NotNull] TextWriter writer)
         {
-            ComponentType = componentType;
+            ViewComponentDescriptor = viewComponentDescriptor;
+            Arguments = arguments;
             ViewContext = viewContext;
             Writer = writer;
         }
 
-        public TypeInfo ComponentType { get; private set; }
+        /// <summary>
+        /// Gets the View Component arguments. 
+        /// </summary>
+        public object[] Arguments { get; }
 
-        public ViewContext ViewContext { get; private set; }
+        /// <summary>
+        /// Gets the <see cref="ViewComponentDescriptor"/> for the View Component being invoked.
+        /// </summary>
+        public ViewComponentDescriptor ViewComponentDescriptor { get; }
 
-        public TextWriter Writer { get; private set; }
+        /// <summary>
+        /// Gets the <see cref="ViewContext"/>.
+        /// </summary>
+        public ViewContext ViewContext { get; }
+
+        /// <summary>
+        /// Gets the <see cref="TextWriter"/> for writing output.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="IViewComponentHelper.Invoke(string, object[])"/> or a similar overload is used to invoke the
+        /// View Component, then <see cref="Writer"/> will be different than <see cref="ViewContext.Writer"/>.
```

Comment:
I think it's fine. The summary tells you what to do (use *this* one). Most users will **never** use any of this stuff anyway, they will return a result.